cardano-cli/src/Cardano/CLI/Read.hs

```diff
@@ -632,16 +628,16 @@ acceptKeyWitnessCDDLSerialisation err =
 
 readCddlWitness
   :: FilePath
-  -> IO (Either (FileError TextEnvelopeCddlError) CddlWitness)
+  -> IO (Either (FileError TextEnvelopeCddlError) (InAnyShelleyBasedEra KeyWitness))
 readCddlWitness fp = do
   readFileTextEnvelopeCddlAnyOf teTypes fp
  where
-  teTypes = [ FromCDDLWitness "TxWitness ShelleyEra" CddlWitness
-            , FromCDDLWitness "TxWitness AllegraEra" CddlWitness
-            , FromCDDLWitness "TxWitness MaryEra" CddlWitness
-            , FromCDDLWitness "TxWitness AlonzoEra" CddlWitness
-            , FromCDDLWitness "TxWitness BabbageEra" CddlWitness
-            , FromCDDLWitness "TxWitness ConwayEra" CddlWitness
+  teTypes = [ FromCDDLWitness "TxWitness ShelleyEra" coerce
```

Comment:
Why do we need `coerce` here? Isn't that `InAnyShelleyBasedEra KeyWitness -> InAnyShelleyBasedEra KeyWitness`, so `id`?

Reply:
It seems that it is no longer necessary in my new approach. I am not sure how that worked, TBH, it was trial and error...